src/stdlib_io_npy_save.fypp

```diff
@@ -128,7 +128,9 @@ contains
         end if
 
         if (present(iomsg)) then
-            iomsg = "Failed to write array to file '"//filename//"'"
+            if (stat /= 0) then
```

Comment:
Why not check for `present(iomsg) .and. stat /= 0`?

Reply:
I did that initially, but then decided to go with nested if's to avoid the `.and.`.